update-kernel-versions.sh

```diff
@@ -15,9 +15,53 @@ kernel_tags=$(git -c 'versionsort.suffix=-' \
 source linux-tkg-config/prepare
 trap - EXIT
 
+# Start by making sure our GnuPG environment is sane
+if [[ ! -x /usr/bin/gpg ]]; then
+  echo "Could not find gpg"
+  exit 4
+fi
+
+## Generate the keyring
+if [ ! -s gnupg/keyring.gpg ]; then
+  if [[ ! -d gnupg ]]; then
+    echo "gnupg directory does not exist"
+    mkdir -p -m 0700 gnupg
+  fi
+  echo "Making sure we have all the necessary keys"
+  gpg --batch --quiet --homedir gnupg --auto-key-locate wkd --locate-keys torvalds@kernel.org gregkh@kernel.org autosigner@kernel.org
+  if [[ $? != "0" ]]; then
+    echo "FAILED to retrieve keys"
+    exit 3
+  fi
+  gpg --batch --homedir gnupg --export torvalds@kernel.org gregkh@kernel.org autosigner@kernel.org > gnupg/keyring.gpg
+fi
+
+# Cleanup
+rm -f v*.x.sha256sums{,.asc}
+
 updates=""
 for _key in "${_current_kernels[@]}"; do
   latest_full_ver=$(echo "$kernel_tags" | grep -F "v$_key" | tail -1 | cut -c2-)
+  kver_major="$(echo ${_key} | cut -d. -f1)"
+  kbasever="$(echo ${_key} | tr -d ".")"
+
+  ## Getting sha256sums by sha256sums.asc
+  if [ ! -s v${kver_major}.x.sha256sums ]; then
+    echo "Downloading the checksums file for linux-v${kver_major}.x"
+    curl -sL -o "v${kver_major}.x.sha256sums.asc" https://cdn.kernel.org/pub/linux/kernel/v${kver_major}.x/sha256sums.asc
+    if [[ $? != "0" ]]; then
+      echo "FAILED to download the v${kver_major}.x checksums file"
+      exit 3
+    fi
+    echo "Verifying the v${kver_major}.x checksums file"
+    count_gpg=$(gpg --homedir gnupg --keyring=gnupg/keyring.gpg --status-fd=1 v${kver_major}.x.sha256sums.asc | grep -c -E '^\[GNUPG:\] (GOODSIG|VALIDSIG)')
+    if [[ ${count_gpg} -lt 2 ]]; then
```

Comment:
Here, if it fails, I think it's better to re-download the shasum file and its signature again, then test again. 

But you know what ? Let's see if it will actually fail in the upcoming updates. If it fails too frequently, we can do the change. Also, the CI run is run every six hours, so if it fails, in the next run it probably won't. But people will have to wait 6 more hours :fearful: xD

Reply:
Hmm, Yea, I would come back to this part if there will be any often problem with this. And maybe would make the script re-invoke via Github workflow somehow. So that all exits from the script with error 3 are simply given a second chance by re-running the script.

Reply:
about 6 hours: Also, I have not yet found a way to replace the scheduled launch of the script. I thought there would be some ready-made option for Github, but maybe something can be done with tracking release tags, for example via email or something, and then triggering an event by writing a comment in the issue/discossion.

Reply:
Let's merge this PR then we can think of this improvement in another